ocs_ci/deployment/factory.py

```diff
@@ -15,7 +15,11 @@ class DeploymentFactory(object):
     def __init__(self):
         # A map all existing deployments and respective classes
         # should be put here
-        self.cls_map = {'aws_ipi': AWSIPI, 'aws_upi': AWSUPI, 'vsphere_upi': VSPHEREUPI}
+        self.cls_map = {
+            'aws_ipi': AWSIPI, 'aws_upi': AWSUPI,
+            'vsphere_upi': VSPHEREUPI,
+            'aws_upi_rhel_workers': AWSUPIRHELWORKERS
```

Comment:
Can we have this rhel workers aligned in get_deployment from different variable and not mess the deployment type which is already used in some places and suppose to be only UPI or IPI?

Reply:
@petr-balogh ,
This one I am not able to comeup with something elegant. Only way I can think is, move this whole RHEL worker code inside AWSUPI class and handle special condition with a flag, but RHEL worker code looks so bloated to me that I felt a separate class is required. I inspected get_deployment() function but couldn't think how to get to this AWSUPIRHELWORKERS class without a map.

Any suggestions you have ?

Reply:
I'm also not a huge fan of having an entirely new class for AWS UPI but with RHEL workers. You have isolated a lot of the code into their own methods so I don't think it would be too much to add to the existing AWSUPI class. We could just have the logic in `deploy_ocp` check for a different configuration key that says we want to use rhel workers and then make the call to `self.add_rhel_workers()`. See my other comment regarding overriding `deploy_ocp` instead of `deploy_cluster`.

Reply:
Done, I have moved everything inside AWSUPI class.